numpy/random/_common.pyx

```diff
@@ -232,13 +232,34 @@ cdef validate_output_shape(iter_shape, np.ndarray output):
         )
 
 
-cdef check_output(object out, object dtype, object size):
+cdef check_output(object out, object dtype, object size, bint require_c_array):
+    """
+    Check user-supplied output array properties and shape
+    
+    Parameters
+    ----------
+    out : {ndarray, None}
+        The array to check.  If None, returns immediately.
+    dtype : dtype
+        The required dtype of out.
+    size : {None, int, tuple[int]}
+        The size passed.  If out is an ndarray, verifies that the shape of out
+        matches size.
+    require_c_array : bool
+        Whether out must be a C-array.  If False, out can be either C- or F-
+        ordered.  If True, must be C-ordered. In either case, must be
+        contiguous, writable and aligned.
+    """
     if out is None:
         return
     cdef np.ndarray out_array = <np.ndarray>out
-    if not (np.PyArray_CHKFLAGS(out_array, np.NPY_CARRAY) or
-            np.PyArray_CHKFLAGS(out_array, np.NPY_FARRAY)):
-        raise ValueError('Supplied output array is not contiguous, writable or aligned.')
+    if not (np.PyArray_ISCARRAY(out_array) or
```

Comment:
I checked a bit more, the `PyArray_IS*` functions also require not swapped, which is in addition to the other requirements. Does that cause a problem?

Reply:
Looks like the random generator doesn't support swapped types anyway, but that was caught in the dtype check.

Reply:
I think we can just add "not swapped" to the error message.

Reply:
I added in machine byte-order, which mirrors the docs for behaved.